### ISSUE TITLE
Upgrade to base image eclipse-temurin:21-jre-noble

### ISSUE THREAD

**Issue by loicmathieu** [kestra-io/kestra#4476]
### Issue description

We had to fix the base image as `eclipse-temurin:21-jre-jamy` as since Ubuntu 24.04 (noble) there is a new `ubuntu` user that uses the user id `1000` which was previously used by the kestra user.

As in our Helm chart we set the user `1000` in the security context this cannot work, see [here](https://github.com/kestra-io/helm-charts/blob/master/charts/kestra/values.yaml#L260). 

We should switch to  `eclipse-temurin:21-jre-noble` to be on the latest Ubuntu version.

**Comment by AlapMistry** [comment]
Hi @loicmathieu,
I can see Kestra repo has `eclipse-temurin:21-jre-jammy`:
https://github.com/kestra-io/kestra/blob/6473a486559b6e308b3cb77ff0031b946957af4d/Dockerfile#L1

I can change it to `eclipse-temurin:21-jre-noble` and run Kestra using Docker compose.
Am I on the right track?

If yes, then please assign it to me.

**Comment by loicmathieu** [comment]
@AlapMistry yes but then something must be found to avoid clash with our Helm chart as the new image will use the same user of pid 1000.

This is not an easy task.

**Comment by MilosPaunovic** [comment]
Hey @AlapMistry, are there any updates on this?

**Comment by AlapMistry** [comment]
Hi @MilosPaunovic,
I changed in the Dockerfile and created an image, but I got '"app/kestra" not found' error while running the container.
I think, it is due to the local build of Kestra. I haven't build Kestra locally.
I checked the Makefile and ran the command accordingly (make build-docker), but it is also running with an error.
Could you guide me to build Kestra in local with Makefile?

Besides, I checked this issue with these `eclipse-temurin` images:
https://stackoverflow.com/questions/78782051/issue-adding-user-in-docker-with-eclipse-temurin-17-jre-image
https://github.com/adoptium/containers/issues/606

There is no change in `adduser` and `addgroup` because this repo already using `useradd` and `groupadd`.
I am checking errors for make commands to build docker image locally.

I am also aware that, Kestra helm charts using `1000` as user ID and group ID. `eclipse-temurin:21-jre-noble` has `ubuntu` user with same ID. Do we have plan to change it for `kestra` user?
https://github.com/kestra-io/helm-charts/blob/47949adaf4f95ea11941e1ac305e52962e76f5b8/charts/kestra/values.yaml#L285

PS: I am able to run Kestra using Docker Compose and manual commands of `Makefile`. I am setting up K8s cluster in my local environment to install Kestra helm chart.

**Comment by MilosPaunovic** [comment]
Maybe @loicmathieu could provide better pointers in this instance.

**Comment by loicmathieu** [comment]
> I am also aware that, Kestra helm charts using 1000 as user ID and group ID. eclipse-temurin:21-jre-noble has ubuntu user with same ID. Do we have plan to change it for kestra user?

Yes, this is the issue.
But as other users may use their own Helm chart or another way of deploying we don't want to change that, it would be a breaking change.

Overall, there are no issue fix for that.

**Comment by AlapMistry** [comment]
Hi @loicmathieu,
I installed the Kestra Helm chart with a changed Docker image. The `kestra` user is created with `1001` ID.

Suggestion: We can create an init container to change `1000` to `1001` for existing deployments. We can also give this ID as configurable and default to `2000` (any number).

It is a breaking change, as you mentioned. It would target the major version of the Kestra Helm chart.

**Comment by MilosPaunovic** [comment]
Hey @AlapMistry, are there any updates on this?

**Comment by AlapMistry** [comment]
Hi @MilosPaunovic and @loicmathieu ,
Do we require to implement as I suggested in my last comment?

**Comment by loicmathieu** [comment]
I think there is no easy solution and as a technical decision should be taken it should be done by someone from the Kestra core team in conjonction with our devops team.

**Comment by loicmathieu** [comment]
Done in https://github.com/kestra-io/kestra/pull/15210
